code/src/components/NotFound.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+
+const NotFound = ({StyledButton})=> {
+    const navigate = useNavigate();
+    const onHomeButtonClick = ()=> {
+        navigate('/')
+    }
+    return (
+        <div className='error'>
+            <p>Sorry, page not found!</p>
+            <StyledButton className='styled-button' onClick={onHomeButtonClick}>Return to Home Page</StyledButton>
+        </div>
+    )

```

Comment:
Really nice not found page - only issue is it only works sometimes, and I'm not completely sure why. Because I can see in the console log that when I'm typing in a bunch of things, it's returning a "Failed to load resource: the server responded with a status of 404 ()" but it's not being redirected to this page.